pkg/v3/tickers/recovery.go

```diff
@@ -1,19 +1,28 @@
 package tickers
 
 import (
+	"sync"
 	"time"
 
 	ocr2keepers "github.com/smartcontractkit/ocr2keepers/pkg"
 )
 
 type recoveryTicker struct {
 	*retryTicker
-	// lastBlock ocr2keepers.BlockKey
+	lock      sync.RWMutex
+	lastBlock ocr2keepers.BlockKey
 }
 
-// getterFn is a function that retrieves the recoveryTick for the given time.
+// setBlock updates the block of the given payload
+func (rt *recoveryTicker) setBlock(p ocr2keepers.UpkeepPayload) {
+	rt.lock.Lock()
+	defer rt.lock.Unlock()
+	p.CheckBlock = rt.lastBlock
+}
+
+// payloadModFunc is a function that updates the payload
 func (rt *recoveryTicker) payloadModFunc(p ocr2keepers.UpkeepPayload) ocr2keepers.UpkeepPayload {
-	// TODO: update block in payload
+	rt.setBlock(p)
```

Comment:
We don't need to use this function here. You can just update the upkeep payload block number directly and then return the modified payload. Ensure you use RLock for this.